src/nodes.coffee

```diff
@@ -1454,10 +1454,10 @@ exports.Value = class Value extends Base
         return new If new Existence(fst), snd, soak: on
       no
 
-  eachName: (iterator) ->
+  eachName: (iterator, {checkAssignability = yes} = {}) ->
```

Comment:
We don't want `Value::eachName()` to always check that it's assignable

Specifically, `for`'s with "unassignable" loop variables were failing here during AST generation, eg
```
for [a..., b] in c
  d
```

So during `For` AST generation pass a `checkAssignability: no` flag to `eachName()`